Make test_cpp_extensions_aot handle lack of pytest more gracefully

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#53740 Make test_cpp_extensions_aot handle lack of pytest more gracefully**

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D26956603](https://our.internmc.facebook.com/intern/diff/D26956603)

ezyang committed:
Make test_cpp_extensions_aot handle lack of pytest more gracefully

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

rgommers reviewed (approved):
Looks great, thanks @ezyang

ezyang committed:
Update on "Make test_cpp_extensions_aot handle lack of pytest more gracefully"


Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D26956603](https://our.internmc.facebook.com/intern/diff/D26956603)

[ghstack-poisoned]
